### PR TITLE
Upgrade to GTK3 and .NET6 (Pinta 2.0)

### PR DESCRIPTION
**Pull request by darkdragon-001**
Build on upstream Pinta efforts.

Fixes the following bugs:
- [x] flathub/flathub#2180

### CODE REVIEW

**Comment by darkdragon-001** [comment]
**Note** In case someone get such warnings after upgrade to Gnome 40 as in https://github.com/flathub/fr.handbrake.ghb/pull/11#issuecomment-808918373, we should add `--filesystem=xdg-run/gvfsd`.

File: com.github.PintaProject.Pinta.yaml [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
 command: pinta
 rename-appdata-file: pinta.appdata.xml
 rename-desktop-file: pinta.desktop
 rename-icon: pinta
 copy-icon: true
 sdk-extensions:
-  - org.freedesktop.Sdk.Extension.mono6
+  - org.freedesktop.Sdk.Extension.dotnet5
 finish-args:
   # X11 + XShm access
```
Comment (hfiguiere) [inline com.github.PintaProject.Pinta.yaml:13]: if you upgrade to Gtk3, it's probably a good idea to add `--socket=wayland` and replace `--socket=x11` to `--socket=fallback-x11`.

Comment (darkdragon-001) [reply com.github.PintaProject.Pinta.yaml:13]: Thanks for the hint! I updated it accordingly.

File: Pinta.yaml [hunk 2]
```diff
@@ -1,14 +1,24 @@
...
-    tag: "1.7"
-    commit: 3d6c188680a389c1f9f40138c092b05ba4009b2e
-  # https://github.com/PintaProject/Pinta/pull/140
-  # https://github.com/PintaProject/Pinta/pull/142
-  # https://github.com/PintaProject/Pinta/pull/145
-  - type: patch
-    path: pinta-1.7-appdata.patch
+    #tag: "1.8"
+    #commit: 3d6c188680a389c1f9f40138c092b05ba4009b2e
+    branch: "gtk3-v2"
```
Comment (hfiguiere) [inline Pinta.yaml]: please pin a commit. A bare branch doesn't lead to a reproducable build. (and comment off branch so that it's not breaking either)

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: Pinta.yaml (hunk 2 above)
Comment (darkdragon-001) [reply Pinta.yaml]: I am waiting for the release of Pinta 1.8 to do exactly this. I kept the comments for a reminder.

Comment (cameronwhite) [reply Pinta.yaml]: FYI the gtk3-v2 branch was recently merged into master, and was upgraded to .NET 6 so a couple updates might be needed here.

Comment (darkdragon-001) [reply Pinta.yaml]: @cameronwhite Thanks for your info. This makes implementing such changes a lot easier when I don't have to figure everything out on my own! See #25 for the changes.

**Comment by darkdragon-001** [comment]
@cameronwhite Can you please test the build and give me the okay to merge to master?

**Comment by cameronwhite** [comment]
👍 did a quick test of the build and it looks good to me

**Comment by darkdragon-001** [comment]
@hfiguiere can you delete the beta branch please?

**Comment by hfiguiere** [comment]
no, i can't. don't have permissions.

**Comment by darkdragon-001** [comment]
@hfiguiere who has permissions?

**Comment by hfiguiere** [comment]
@barthalion or @nedrichards AFAIK.

**Comment by barthalion** [comment]
Please open an issue at flathub/flathub and I will handle it eventually.

**Pull request merged**